pkg/internal/controller/controller.go

```diff
@@ -163,9 +163,7 @@ func (c *Controller) Start(ctx context.Context) error {
 		for _, watch := range c.startWatches {
 			c.Log.Info("Starting EventSource", "source", watch.src)
 
-			watchStartCtx, cancel := context.WithTimeout(ctx, c.CacheSyncTimeout)
-			defer cancel()
-			if err := watch.src.Start(watchStartCtx, watch.handler, c.Queue, watch.predicates...); err != nil {
+			if err := watch.src.Start(ctx, watch.handler, c.Queue, watch.predicates...); err != nil {
```

Comment:
Unfortunately, what we do today is the correct behaviour for `source.Kind`, it has a different interpretation of the purpose of its `ctx` argument. Let's merge this as `source.Channel` doesn't work at all currently, I will open a n issue so we can clarify our interface definitions and update one or the other accordingly

Reply:
(This change will break the waitForSync functionality I believe)